Library/Homebrew/cmd/update-report.rb

```diff
@@ -39,13 +39,23 @@ def update_report_args
   end
 
   def update_report
+    # Avoid mandatory reinstalls of `pkg-config` upon OS upgrade.
+    symlink_current_pkgconfig_directory
     return output_update_report if $stdout.tty?
 
     redirect_stdout($stderr) do
       output_update_report
     end
   end
 
+  def symlink_current_pkgconfig_directory
+    return if OS.linux?
+
+    pkgconfig_dir = HOMEBREW_LIBRARY/"Homebrew/os/mac/pkgconfig"
+    FileUtils.rm_f pkgconfig_dir/"current"
+    pkgconfig_dir.install_symlink MacOS.version => "current"
+  end
```

Comment:
This actually feels a bit more out of place for me here, since `brew.sh` seems more reasonable for setup code that makes sure `brew` is working properly.

Reply:
This seems like a reasonable place to me. I'd suggest `symlink_current_pkgconfig_directory` be pulled into a helper function, though, and it also be called in `install.rb` or `formula_installer.rb` because you cannot rely on all users running `brew update`. It feels like it also warrants a `brew doctor` if missing.

Reply:
> I'd suggest `symlink_current_pkgconfig_directory` be pulled into a helper function, though, and it also be called in `install.rb` or `formula_installer.rb` because you cannot rely on all users running `brew update`. It feels like it also warrants a `brew doctor` if missing.

Yes, agreed.

Reply:
Could actually even be a good candidate for one of the `diagnostic` pre-install mandatory checks that just refuses to continue if it's broken, runs as part of `brew doctor` and we can tell people to run `brew update` or whatever to fix.

Reply:
Added symlink creation to `install.rb` and a diagnostic.

> Could actually even be a good candidate for one of the `diagnostic` pre-install mandatory checks that just refuses to continue if it's broken

I contemplated doing this, but am ambivalent. If @Bo98 thinks it's a good idea then I'm happy to do it though.